test_runner/fixtures/neon_fixtures.py

```diff
@@ -926,6 +968,53 @@ def __exit__(
             if cleanup_error is not None:
                 cleanup_error = e
 
+        if self.test_cgroup_dir is not None:
+            log.info(f"check test runner cgroup for leaked processes: {self.test_cgroup_dir}")
+            procs_file = self.test_cgroup_dir / "cgroup.procs"
+            mypid = os.getpid()
+            # move ourselves out of cgroup
+            with open(self.test_cgroup_dir.parent / "cgroup.procs", "a") as f:
+                f.write(f"{mypid}")
+            # now freeze the test cgroup, so we can race-free list & SIGKILL the processes
+            freeze_file = self.test_cgroup_dir / "cgroup.freeze"
+            freeze_file.write_text("1")
```

Comment:
Alternatively before freezing we could try dumping all of the stacktraces for the remaining processes with gdb, if needed, and that's all which would need to change? Or would you freeze, inspect pids, thaw then try to dump each, then re-freeze, inspect, kill, thaw, wait?

Reply:
can we close this / convert into a follow-up for the parent epic https://github.com/neondatabase/neon/issues/6485

Reply:
https://github.com/neondatabase/neon/issues/6489 done, resolving.